### PR TITLE
(PUP-8321) add translations for deprecation_warning messages

### PR DESCRIPTION
**Pull request by ericldelaney**

### CODE REVIEW

File: lib/puppet/agent/locker.rb [hunk 1]
```diff
@@ -29,11 +29,11 @@ def lock
...
-    Puppet.deprecation_warning _(<<-ENDHEREDOC)
-Puppet::Agent::Locker.running? is deprecated as it is inherently unsafe.
-The only safe way to know if the lock is locked is to try lock and perform some
-action and then handle the LockError that may result.
-ENDHEREDOC
+    #TRANSLATORS 'Puppet::Agent::Locker.running?' is a method name and should not be translated
+    message = _('Puppet::Agent::Locker.running? is deprecated as it is inherently unsafe.')
+    message += ' ' + _('The only safe way to know if the lock is locked is to try lock and perform some'\
+                       'action and then handle the LockError that may result.')
+    Puppet.deprecation_warning _(message)
```
Comment (melissa) [inline lib/puppet/agent/locker.rb]: do you still need `_()` to wrap message here? I was under the impression the previous lines negated the need for this?

File: lib/puppet/agent/locker.rb [hunk 2]
```diff
@@ -29,11 +29,11 @@ def lock
...
   def running?
-    Puppet.deprecation_warning _(<<-ENDHEREDOC)
-Puppet::Agent::Locker.running? is deprecated as it is inherently unsafe.
-The only safe way to know if the lock is locked is to try lock and perform some
-action and then handle the LockError that may result.
-ENDHEREDOC
+    #TRANSLATORS 'Puppet::Agent::Locker.running?' is a method name and should not be translated
+    message = _('Puppet::Agent::Locker.running? is deprecated as it is inherently unsafe.')
+    message += ' ' + _('The only safe way to know if the lock is locked is to try lock and perform some'\
+                       'action and then handle the LockError that may result.')
```
Comment (melissa) [inline lib/puppet/agent/locker.rb]: It looks like we don't want to translate `LockError` either

File: lib/puppet/parameter/value_collection.rb [hunk 3]
```diff
@@ -133,7 +133,9 @@ def newvalue(name, options = {}, &block)
     call_opt = options[:call]
     unless call_opt.nil?
       devfail "Cannot use obsolete :call value '#{call_opt}' for property '#{self.class.name}'" unless call_opt == :none || call_opt == :instead
-      Puppet.deprecation_warning(_("Property option :call is deprecated and no longer used. Please remove it."))
+      message = _("Property option :call is deprecated and no longer used.")
```
Comment (melissa) [inline lib/puppet/parameter/value_collection.rb]: `:call` probably shouldn't be translated here

File: lib/puppet/settings.rb [hunk 4]
```diff
@@ -1158,9 +1158,13 @@ def issue_deprecation_warning(setting, msg = nil)
       msg << " #{ref}" if ref
       Puppet.deprecation_warning(msg)
     when setting.completely_deprecated?
-      Puppet.deprecation_warning("Setting #{name} is deprecated. #{ref}", "setting-#{name}")
+      message = _("Setting %{name} is deprecated.") % { name: name }
+      message += " #{ref}"
+      Puppet.deprecation_warning(message, "setting-#{name}")
     when setting.allowed_on_commandline?
-      Puppet.deprecation_warning("Setting #{name} is deprecated in puppet.conf. #{ref}", "puppet-conf-setting-#{name}")
+      message = _("Setting %{name} is deprecated in puppet.conf.") % { name: name }
```
Comment (melissa) [inline lib/puppet/settings.rb]: `puppet.conf` is a filename and shouldn't be translated

File: lib/puppet/agent/locker.rb (hunk 1 above)
Comment (ericldelaney) [reply lib/puppet/agent/locker.rb]: Fixed

File: lib/puppet/settings.rb (hunk 4 above)
Comment (ericldelaney) [reply lib/puppet/settings.rb]: fixed

File: lib/puppet/parameter/value_collection.rb (hunk 3 above)
Comment (ericldelaney) [reply lib/puppet/parameter/value_collection.rb]: fixed

File: lib/puppet/agent/locker.rb (hunk 2 above)
Comment (ericldelaney) [reply lib/puppet/agent/locker.rb]: fixed

**Comment by ericldelaney** [comment]
@melissa Updated

File: lib/puppet/agent/locker.rb [hunk 5]
```diff
@@ -29,11 +29,12 @@ def lock
...
   def running?
-    Puppet.deprecation_warning _(<<-ENDHEREDOC)
-Puppet::Agent::Locker.running? is deprecated as it is inherently unsafe.
-The only safe way to know if the lock is locked is to try lock and perform some
-action and then handle the LockError that may result.
-ENDHEREDOC
+    #TRANSLATORS 'Puppet::Agent::Locker.running?' is a method name and should not be translated
+    message = _('Puppet::Agent::Locker.running? is deprecated as it is inherently unsafe.')
+    #TRANSLATORS 'LockError' should not be translated
+    message += ' ' + _('The only safe way to know if the lock is locked is to try lock and perform some'\
```
Comment (8675309) [inline lib/puppet/agent/locker.rb]: missing space at end of line after 'some'

**Comment by ericldelaney** [comment]
@8675309 Updated

**Comment by jtappa** [comment]
Just waiting for this to get through Travis

**Pull request merged**